crates/bevy_reflect/bevy_reflect_derive/src/lib.rs

```diff
@@ -60,11 +142,30 @@ pub fn derive_reflect(input: TokenStream) -> TokenStream {
 
 /// Derives the `FromReflect` trait.
 ///
-/// This macro supports the following field attributes:
-/// * `#[reflect(ignore)]`: Ignores the field. This requires the field to implement [`Default`].
-/// * `#[reflect(default)]`: If the field's value cannot be read, uses its [`Default`] implementation.
-/// * `#[reflect(default = "some_func")]`: If the field's value cannot be read, uses the function with the given name.
+/// # Field Attributes
+///
+/// ## `#[reflect(ignore)]`
 ///
+/// The `#[reflect(ignore)]` attribute is shared with the [`#[derive(Reflect)]`](Reflect) macro and has much of the same
+/// functionality in that it marks a field to be ignored by the reflection API.
+///
+/// The only major difference is that using it with this derive requires that the field implements [`Default`].
```

Comment:
I wanted to resolve the Fixme here: https://github.com/bevyengine/bevy/blob/main/crates/bevy_asset/src/handle.rs#L119

Initially, I was confused why it works, as when expanding the derive, the Default of ` Handle<T>` was used.

Reading the following from the `Reflect` documentation of this PR removed the confusion:
```
/// * `#[reflect(Default)]` will register the `ReflectDefault` type data as normal.
///   However, it will also affect how certain other operations are performed in order
///   to improve performance and/or robustness.
///   An example of where this is used is in the [`FromReflect`] derive macro,
///   where adding this attribute will cause the `FromReflect` implementation to create
///   a base value using its [`Default`] implementation avoiding issues with ignored fields.
```
I don't know how to properly write this down here, but probably something like this:

```suggestion
/// The only major difference is that using it with this derive requires that the field implements [`Default`] or the struct to have a `#[reflect(Default)]`.
```